source/types.ts

```diff
@@ -0,0 +1,233 @@
+
+import {JSONSchema as TypedJSONSchema} from 'json-schema-typed';
+// eslint-disable unicorn/import-index
+import {Conf} from '.';
+import {EventEmitter} from 'events';
+
+export interface Options<T> {
+	/**
+	Config used if there are no existing config.
+
+	**Note:** The values in `defaults` will overwrite the `default` key in the `schema` option.
+	*/
+	defaults?: Readonly<T>;
+
+	/**
+	[JSON Schema](https://json-schema.org) to validate your config data.
+
+	Under the hood, the JSON Schema validator [ajv](https://github.com/epoberezkin/ajv) is used to validate your config. We use [JSON Schema draft-07](http://json-schema.org/latest/json-schema-validation.html) and support all [validation keywords](https://github.com/epoberezkin/ajv/blob/master/KEYWORDS.md) and [formats](https://github.com/epoberezkin/ajv#formats).
+
+	You should define your schema as an object where each key is the name of your data's property and each value is a JSON schema used to validate that property. See more [here](https://json-schema.org/understanding-json-schema/reference/object.html#properties).
+
+	@example
+	```
+	import Conf = require('conf');
+
+	const schema = {
+		foo: {
+			type: 'number',
+			maximum: 100,
+			minimum: 1,
+			default: 50
+		},
+		bar: {
+			type: 'string',
+			format: 'url'
+		}
+	};
+
+	const config = new Conf({schema});
+
+	console.log(config.get('foo'));
+	//=> 50
+
+	config.set('foo', '1');
+	// [Error: Config schema violation: `foo` should be number]
+	```
+
+	**Note:** The `default` value will be overwritten by the `defaults` option if set.
+	*/
+	schema?: Schema<T>;
+
+	/**
+	Name of the config file (without extension).
+
+	Useful if you need multiple config files for your app or module. For example, different config files between two major versions.
+
+	@default 'config'
+	*/
+	configName?: string;
+
+	/**
+	You only need to specify this if you don't have a package.json file in your project or if it doesn't have a name defined within it.
+
+	Default: The name field in the `package.json` closest to where `conf` is imported.
+	*/
+	projectName?: string;
+
+	/**
+	You only need to specify this if you don't have a package.json file in your project or if it doesn't have a version defined within it.
+
+	Default: The name field in the `package.json` closest to where `conf` is imported.
+	*/
+	projectVersion?: string;
+
+	/**
+	_Don't use this feature until [this issue](https://github.com/sindresorhus/conf/issues/92) has been fixed._
```

Comment:
This was removed. Can you go through each commit since this was opened to ensure that it's fully in sync with `master`?

Reply:
I did already go through everything on master. I just did not check the `index.d.ts` since it is auto generated on my end. I'll recheck to make sure, but it should be good besides that.